Sming/SmingCore/Network/Http/HttpRequest.h

```diff
@@ -30,82 +31,157 @@ typedef Delegate<int(HttpConnection& client, HttpResponse& response)> RequestHea
 typedef Delegate<int(HttpConnection& client, const char* at, size_t length)> RequestBodyDelegate;
 typedef Delegate<int(HttpConnection& client, bool successful)> RequestCompletedDelegate;
 
+/*
+ * Encapsulates an incoming or outgoing request
+ */
 class HttpRequest
 {
 	friend class HttpClient;
 	friend class HttpConnection;
 	friend class HttpServerConnection;
 
 public:
-	HttpRequest(const URL& uri);
+	HttpRequest()
+	{
+	}
+
+	HttpRequest(const URL& uri) : uri(uri)
+	{
+	}
+
 	HttpRequest(const HttpRequest& value);
-	__forceinline HttpRequest* clone() const
+
+	HttpRequest* clone() const
 	{
 		return new HttpRequest(*this);
 	}
+
 	HttpRequest& operator=(const HttpRequest& rhs);
-	~HttpRequest();
 
-	HttpRequest* setURL(const URL& uri);
+	~HttpRequest()
+	{
+		reset();
+	}
+
+	HttpRequest* setURL(const URL& uri)
+	{
+		this->uri = uri;
+		return this;
+	}
+
+	HttpRequest* setMethod(HttpMethod method)
+	{
+		this->method = method;
+		return this;
+	}
 
-	HttpRequest* setMethod(const HttpMethod method);
+	HttpRequest* setHeaders(const HttpHeaders& headers)
+	{
+		this->headers.setMultiple(headers);
+		return this;
+	}
 
-	HttpRequest* setHeaders(const HttpHeaders& headers);
+	HttpRequest* setHeader(const String& name, const String& value)
+	{
+		headers[name] = value;
+		return this;
+	}
 
-	HttpRequest* setHeader(const String& name, const String& value);
+	/** @deprecated this replaces existing post parameter set, which is different to
```

Comment:
Can you add here also a very small example of the new recommended way? That target audience is beginners.

Reply:
setPostParameters() isn't actually used anywhere in Sming or its samples. Would it be safe to delete?

Reply:
If we're keeping it, I'd probably amend the comment to something like:
```
To set all post parameters, use request.postParams = params
To copy additional post parameters, use request.postParams.setMultiple(params)
```